streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -223,10 +222,9 @@ public void shouldSendDataToTopicUsingProduced() {
     }
 
     @Test
-    // TODO: this test should be refactored when we removed KStreamBuilder so that the created Topology contains internal topics as well
     public void shouldUseRecordMetadataTimestampExtractorWhenInternalRepartitioningTopicCreated() {
-        final KStreamBuilder builder = new KStreamBuilder();
-        KStream<String, String> kStream = builder.stream(Serdes.String(), Serdes.String(), "topic-1");
+        final StreamsBuilder builder = new StreamsBuilder();
+        KStream<String, String> kStream = builder.stream("topic-1", stringConsumed);
```

Comment:
nit: add `final` (can we clean up the whole method and add `final` everywhere?)

Reply:
ack.